Process.c

```diff
@@ -253,35 +256,374 @@ void Process_fillStarttimeBuffer(Process* this) {
+         (void)stpcpyWithNewlineConversion(str, cmdline + cmdlineBasenameStart);
+      }
+
+      return;
+   }
+
+   int exeLen = strlen(this->procExe);
+   int exeBasenameOffset = this->procExeBasenameOffset;
+   int exeBasenameLen = exeLen - exeBasenameOffset;
+
+   assert(exeBasenameOffset >= 0);
+   assert(exeBasenameOffset <= (int)strlen(procExe));
+
+   bool haveCommInExe = false;
+   if (procExe && procComm) {
+      haveCommInExe = strncmp(procExe + exeBasenameOffset, procComm, TASK_COMM_LEN - 1) == 0;
+   }
+
+   /* Start with copying exe */
+   if (showProgramPath) {
+      if (haveCommInExe)
+         WRITE_HIGHLIGHT(exeBasenameOffset, exeBasenameLen, commAttr, CMDLINE_HIGHLIGHT_FLAG_COMM);
+      WRITE_HIGHLIGHT(exeBasenameOffset, exeBasenameLen, baseAttr, CMDLINE_HIGHLIGHT_FLAG_BASENAME);
+      if (this->procExeDeleted)
+         WRITE_HIGHLIGHT(exeBasenameOffset, exeBasenameLen, delAttr, CMDLINE_HIGHLIGHT_FLAG_DELETED);
+      str = stpcpy(str, procExe);
+   } else {
+      if (haveCommInExe)
+         WRITE_HIGHLIGHT(0, exeBasenameLen, commAttr, CMDLINE_HIGHLIGHT_FLAG_COMM);
+      WRITE_HIGHLIGHT(0, exeBasenameLen, baseAttr, CMDLINE_HIGHLIGHT_FLAG_BASENAME);
+      if (this->procExeDeleted)
+         WRITE_HIGHLIGHT(0, exeBasenameLen, delAttr, CMDLINE_HIGHLIGHT_FLAG_DELETED);
+      str = stpcpy(str, procExe + exeBasenameOffset);
+   }
+
+   bool haveCommInCmdline = false;
+   int commStart = 0;
+   int commEnd = 0;
+
+   /* Try to match procComm with procExe's basename: This is reliable (predictable) */
+   if (searchCommInCmdline) {
+      /* commStart/commEnd will be adjusted later along with cmdline */
+      haveCommInCmdline = findCommInCmdline(procComm, cmdline, cmdlineBasenameStart, &commStart, &commEnd);
+   }
+
+   int matchLen = matchCmdlinePrefixWithExeSuffix(cmdline, cmdlineBasenameStart, procExe, exeBasenameOffset, exeBasenameLen);
+
+   bool haveCommField = false;
+
+   if (!haveCommInExe && !haveCommInCmdline && procComm) {
+      WRITE_SEPARATOR;
+      WRITE_HIGHLIGHT(0, strlen(procComm), commAttr, CMDLINE_HIGHLIGHT_FLAG_COMM);
+      str = stpcpy(str, procComm);
+      haveCommField = true;
+   }
+
+   if (matchLen) {
+      /* strip the matched exe prefix */
+      cmdline += matchLen;
+
+      commStart -= matchLen;
+      commEnd -= matchLen;
+   }
+
+   if (!matchLen || (haveCommField && *cmdline)) {
+      /* cmdline will be a separate field */
+      WRITE_SEPARATOR;
+   }
+
+   if (!haveCommInExe && haveCommInCmdline && !haveCommField)
+      WRITE_HIGHLIGHT(commStart, commEnd - commStart, commAttr, CMDLINE_HIGHLIGHT_FLAG_COMM);
+
+   /* Display cmdline if it hasn't been consumed by procExe */
+   if (*cmdline)
+      (void)stpcpyWithNewlineConversion(str, cmdline);
+
+   #undef WRITE_SEPARATOR
+   #undef WRITE_HIGHLIGHT
+}
+
+void Process_writeCommand(const Process *this, int attr, int baseAttr, RichString *str) {
+   (void)baseAttr;
```

Comment:
Why not remove the function parameter `baseAttr`?

Reply:
Mostly because I'm trying to stick with just moving the code into the right places for now. There's a second set of changes I noticed where some more cleanup of this could be done (but which would delay this PR even further).